### PR TITLE
Add upload test cases for VideoPress block

### PR DESCRIPTION
**Pull request by fluiddot**
**PRs related:**
* https://github.com/WordPress/gutenberg/pull/50398
* https://github.com/WordPress/gutenberg/pull/50399
* https://github.com/Automattic/jetpack/pull/30486

Part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/5734.

This PR adds several test cases to cover the upload functionality of the VideoPress. block.

**To test:**
1. Run the command: `TEST_RN_PLATFORM=ios npm run test`
2. Run the command: `TEST_RN_PLATFORM=android npm run test`
3. Check that all CI jobs succeed.

PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
 		"prewpandroid": "rm -Rf $TMPDIR/gbmobile-wpandroidfakernroot && mkdir $TMPDIR/gbmobile-wpandroidfakernroot && ln -s $(cd \"$(dirname \"../../../\")\"; pwd) $TMPDIR/gbmobile-wpandroidfakernroot/android",
 		"wpandroid": "cd gutenberg && react-native run-android --root $TMPDIR/gbmobile-wpandroidfakernroot --variant wasabiDebug --appIdSuffix beta --appFolder WordPress --main-activity=ui.WPLaunchActivity",
 		"test": "cross-env NODE_ENV=test jest --verbose --config ./jest.config.js",
+		"test:update": "cross-env NODE_ENV=test jest --verbose --config ./jest.config.js --updateSnapshot",
```
Comment (fluiddot) [inline package.json:82]: I added this command to allow Jest snapshots to be updated, similar to the command we have in Gutenberg ([reference](https://github.com/WordPress/gutenberg/blob/05ad081521b040c7ba5ff5726dd16af9523b504e/packages/react-native-editor/package.json#LL109C4-L109C11)).

File: src/test/videopress/upload.js [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+				ios: {
+					id: VIDEOPRESS_GUID,
+				},
+			} ),
+		} );
+
+		// Requests:
+		//  - Token request
+		//  - Metadata request
+		//  - Check ownership request
```
Comment (fluiddot) [inline src/test/videopress/upload.js:232]: Note this request is relatively new and at some point, we'd need to incorporate the logic in the native version (https://github.com/Automattic/jetpack/pull/30438).

**Review by jhnstn — APPROVED** [review]
Tests passed as expected
Nice work!

**Comment by fluiddot** [comment]
@jhnstn since you approved some of the related PRs, could you take a look at https://github.com/WordPress/gutenberg/pull/50398? Once that one is approved we can merge this PR. Thanks 🙇 !

**Pull request merged**